programmer-skills/http-intro.md

```diff
@@ -0,0 +1,361 @@
+Introduction to HTTP
+====================
+
+The Hypertext Transfer Protocol (HTTP) is the foundation of web communication, so some familiarity with it is essential for any web developer. HTTP works on the basis of a *client* sending a *request* to a *server*, which in return provides a *response*. You can think of the client as being the user's web browser and the server as being any kind of back-end server you might write (e.g. in Express.js).
+
+Browsers must formulate valid requests with the appropriate information for that request. Sevrvers will respond to requests according to the specifications. Browsers will in turn interpret these responses according to the specifications. this means that it is important that both your front-end and back-end code is using HTTP correctly.
+
+Of course, as so often happens in web development, different browsers have their own implementation quirks.
+
```

Comment:
Change to "Competing browsers behave differently with their own individual behaviour."